benchmark/main.cpp

```diff
@@ -819,7 +777,7 @@ static int benchmark( int argc, char *argv[], mbedtls_platform_context* ctx )
             mbedtls_snprintf( title, sizeof( title ), "ECDSA-%s",
                                               curve_info->name );
             TIME_PUBLIC( title, "sign",
-                    ret = mbedtls_ecdsa_write_signature( &ecdsa, MBEDTLS_MD_SHA256, buf, curve_info->bit_size,
+                    ret = mbedtls_ecdsa_write_signature( &ecdsa, MBEDTLS_MD_SHA256, buf, ( curve_info->bit_size + 7 ) / 8,
```

Comment:
Was the previous buffer size simply inaccurate, or we don't have so much space? If this amount only plays a role in checking whether the buffer is big enough, why don't we pass the real buffer size?

Reply:
The parameter is `hlen` not buffer length. So, sizeof(buf) is inaccurate.
The previous buffer size was sent as bits, not bytes, but the API expects length in bytes.
SInce this is a benchmark application, it didn't verify with expected result, and the SW implementation succeeded. I noticed this issue, when a HW accelerator expects only specific standard lengths, so it returned an error